src/yggdrasil/tun.go

```diff
@@ -66,7 +66,8 @@ func (tun *tunDevice) read() error {
 	for {
 		n, err := tun.iface.Read(buf)
 		if err != nil {
-			panic(err)
+			// panic(err)
+			return err
```

Comment:
Note to self: don't just `go tun.read()`, since it returns an error. Go some function that catches the error, and can either restart the read if it's recoverable or, signal that there was a failure if it's not (log and/or panic).